random_menu_optionals.rb

```diff
@@ -0,0 +1,79 @@
+
+
+=begin
+This is a random menu generator, it creates three-word names for new foods.
+The program starts with a library of food words contained in one array of
+adjectives, another array of cooking styles, and a third array of foods.
+
+This script can make from 1 to 10 food items.  The user enters the number of
+items they would like to be outputed.
+
+In each iteration one object from each of three arrays is randomly chosen
+and combined with the others to make a single string.  Then, the componets
+of that string are removed from the food word library.  The removal makes
+sure there are no repeated words in the final menu.
+=end
+
+#Set the components of the menu word libraries.
+#Possible improvment would be to let the user enter these
+arr1 = %w(Creamy Yummy Hot Cold Spicy Not-yo-mama's Cheesy Chicago Classic Aged)
+arr2 = %w(pan-fried teriyaki steamed toasted roasted baked balsamic sautéd boiled cured )
+arr3 = %w(clams dumplings fritters burger adobo chili curds pizza soup quesadilla)
+
+
+#Set methods:
+#This method takes one array (arr), compares it to another array (used).
+#If there is a match the matched element is removed from arr.
+#It is used here to remove already selected words so that there is no repetition
+#in the menu.
+def remove (arr, used)
+  used.each do |n|
+    if arr.include?(n.to_s)
+      arr.delete(n.to_s)
+    end
+  end
+end
+
+#This method test if an inputed value is an interger between upper and lower bounds.
+#upper_bound and lower_bound must be integers, otherwise this breaks.
+def valid_int (input, lower_bound, upper_bound)
+  valid_array= Array (lower_bound..upper_bound)
+  s_valid_array= valid_array.map(&:to_s)
+  s_valid_array[2]
+  x=0
+  while x==0
+    if s_valid_array.include?(input)
+      x=1
```

Comment:
Watch your style here - it's typically considered more readable to put space on either side of an operator or equals sign. So `while x == 0` rather than `while x==0`.